packages/teuchos/parameterlist/src/Teuchos_ParameterEntry.hpp

```diff
@@ -260,7 +267,7 @@ class TEUCHOSPARAMETERLIST_LIB_DLL_EXPORT ParameterEntry {
     in the ParameterEntry class, since the user does not have to pass in a pointer of type \c T.
 */
 template<typename T>
-inline T& getValue( const ParameterEntry &entry )
```

Comment:
It just occurred to me that we can't make this change to return by value, at least not in this way.  The reason is that the Teuchos::ParameterList is not only used for small objects but it is also used as a general object data-base and would totally break that usage.

Instead, we need to perhaps come up with a new set of functions that can do auto conversions. I will comment on this in #2559.